### PR TITLE
Site Switcher: Introduce "Manage Sites" button at the bottom

### LINKED ISSUE #67041: Site Switcher: Move 'Manage [N] Sites' button to bottom of the UI, add conditional 'View [feature] for sites' button
After #66804 lands, we'd like to add a 'Manage Sites' button to the bottom of the Site Switcher UI.

image

The number should reflect the user's total number of sites.

Design for the button is a little TBD at the moment.

Next, the existing 'All my sites' button in the Site Switcher is a little erratic: sometimes it takes you to a list of all your sites, and other times it takes you the multi-site view relating to the feature you're looking at (e.g. stats). We'd like to instead change the existing 'All my sites' button to a 'View [feature] for sites' button that only appears when a given view has a multi-site context:

image

**Done is:**
* [x] A new 'Manage [N] Sites' button appears at the bottom of the Site Switcher UI.
* [ ] The count in the button reflects the user's total number of sites.
* [x] The button links to the new Sites page (`/sites`).
* [x] The button triggers a Tracks event when clicked.
* [x] The existing 'All my sites' UI element is removed.
* [x] A 'View [feature] for sites' button appears when a given view has a multi-site context.
* [ ] The 'Add new site' button is styled as depicted (without dropdown, though).
* [ ] Changes are tested against Jetpack Cloud to ensure no regressions.

### PR DESCRIPTION
**Pull request by zaguiini**
**IMPORTANT: This PR does NOT modify the "Add New Site" button inside the Site Switcher. That was addressed in https://github.com/Automattic/wp-calypso/pull/67346.**

#### Proposed Changes

Part of #67041. Let's add a new "Manage Sites" button at the bottom when opening the site switcher. This button will send the user to `/sites` once clicked. It will preserve the search query so they can reproduce that when browsing. A Tracks event `calypso_manage_sites_click` is dispatched whenever this button is clicked: 1045-gh-Automattic/tracks-events-registration.

Also, we now have proper multi-site context labeling:

image

The "All My Sites" label was very misleading and did not indicate that the user will see, for example, all posts for their sites. Now they do. I've updated the tracks event (1043-gh-Automattic/tracks-events-registration) to include the slug for the multi-site context page that the user is seeing.

#### Manage sites button

1. Click `Switch Site`;
2. Check that "Manage Sites" button is stuck to the bottom of the page, even if you scroll the sites list:
image
3. Type any term on the search bar. Check that the number of sites reflects the search results count;
4. Check that the "Manage Sites" button preserves the search query and opens the Sites dashboard with the pre-filled term;
5. Check that a Tracks event called `calypso_manage_sites_click` was dispatched.

#### Multi-site context pages

1. Click `Posts` on any selected site;
2. Click `Switch Site`;
3. Check that the CTA is now aware of the all-sites contextual section:
image
4. Click that button;
6. Check that it's gone when opening the site switcher again. It should not show up when the user is already browsing such kind of page;
7. Check that a Tracks event called `calypso_all_my_sites_click` was dispatched with a `multi_site_context_slug` property.

#### Overall regression testing

Check that Jetpack Cloud (`yarn start-jetpack-cloud`) is unaffected by the changes, neither are`<SiteSelector />` usages across Calypso (e.g. http://calypso.localhost:3000/marketing/tools).

### CODE REVIEW

File: client/components/search-sites/index.js [hunk 1]
```diff
@@ -13,18 +13,19 @@ export default function searchSites( WrappedComponent ) {
 	class Searcher extends Component {
 		state = { term: null };
 
-		setSearchTerm = ( term ) => this.setState( { term } );
+		setSearchTerm = ( term ) => this.setState( { term: term?.toLowerCase() ?? '' } );
 
 		getSearchResults() {
 			return this.state.term
-				? searchCollection( this.props.sites, this.state.term.toLowerCase(), [ 'name', 'URL' ] )
+				? searchCollection( this.props.sites, this.state.term, [ 'name', 'URL' ] )
```
Comment (danielbachhuber) [inline client/components/search-sites/index.js:20]: @zaguiini Think we should add your fuzzy searching in a follow-up PR?

**Review by danielbachhuber — DISMISSED** [review]
Looks solid, nice work 🛠️ 

I've confirmed:
* The "Manage [N] Sites" button appears as expected. Clicking the button triggers the `calypso_manage_sites_click` event.
* The "View [feature] for all sites" button only appears when that feature is present for all sites.

Interestingly, the "Manage [N] Sites" button appears for me in Jetpack Cloud:

image

I started Jetpack Cloud with `yarn start-jetpack-cloud-p`.

Also, this number discrepancy is a little odd:

image

Let's drop the count from the button and use "Manage Sites" instead.

Lastly, let's add the "Add new site" button to this PR, and then we can ship early next week.

File: client/components/search-sites/index.js (hunk 1 above)
Comment (danielbachhuber) [reply client/components/search-sites/index.js:20]: Created #67316

**Comment by zaguiini** [comment]
>Lastly, let's add the "Add new site" button to this PR, and then we can ship early next week.

I think I'll add it as a separate PR, trying to separate concerns here.

**Comment by danielbachhuber** [comment]
> I think I'll add it as a separate PR, trying to separate concerns here.

Sounds good.

**Comment by zaguiini** [comment]
@danielbachhuber, changes made. Could you please retry?

I could not reproduce the `Manage Sites` button on the JC dashboard 😢 It does not happen if I run JC exclusively, neither if I run with Calypso side-by-side, nor in JC live Docker instance in this PR... Got me thinking. Can you still reproduce the problem?

**Comment by p-jackson** [comment]
I think the `Manage sites` button needs to take better account of the current "interface theme".

This is what it looks like when the user has chosen the "Light" theme

CleanShot 2022-09-06 at 15 08 47@2x


I think there should be a "foreground colour" variable that we can use for the text and border colours.

**Comment by p-jackson** [comment]
So I ended up copying the colours that the list items use. The "Manage sites" button sorta looks like a plain text button, and so having it use the same hover effect as the text looks good. Especially for this theme:

Sep-06-2022 15-38-19

It's like it's made for it. Unfortunately not every theme changes the text colour on hover. Some of them set a background colour. So I've done the same since I know each of those colours have been tested in the various themes.

Sep-06-2022 15-37-44
Sep-06-2022 15-37-10

It doesn't look _too_ bad, although the "coloured background with white border" style might be a little unexpected...

I also tried something like `rgba( var( --color-sidebar-text-rgb ), 0.75 )` for the hover style so that we wouldn't have to have the filled-in background for the hover state. This works ok, but it means that for themes like the first GIF above the text hover colour for the button doesn't match the text hover colour for the list items. And that looks a bit peculiar.

File: client/layout/index.jsx [hunk 2]
```diff
@@ -49,9 +49,12 @@ import { handleScroll } from './utils';
...
 import './style.scss';
 
-function SidebarScrollSynchronizer() {
+function SidebarScrollSynchronizer( { layoutFocus } ) {
 	const isNarrow = useBreakpoint( '<660px' );
-	const active = ! isNarrow && ! config.isEnabled( 'jetpack-cloud' ); // Jetpack cloud hasn't yet aligned with WPCOM.
+	const active =
+		! isNarrow &&
+		! config.isEnabled( 'jetpack-cloud' ) && // Jetpack cloud hasn't yet aligned with WPCOM.
+		layoutFocus !== 'sites';
```
Comment (p-jackson) [inline client/layout/index.jsx]: @danielbachhuber do you about why this `SidebarScrollSynchronizer` component needed to be deactivated when in the `sites` layout focus? I removed this line and everything worked as I expected.

File: client/my-sites/sidebar/static-data/all-sites-menu.js [hunk 3]
```diff
@@ -12,41 +12,47 @@ export default function allSitesMenu() {
 			icon: 'dashicons-chart-bar',
 			slug: 'stats',
 			title: translate( 'Stats' ),
+			navigationLabel: translate( 'View stats for all sites' ),
```
Comment (p-jackson) [inline client/my-sites/sidebar/static-data/all-sites-menu.js:15]: I've added individual translations for each of the "all sites" links. Previously we were using string interpolation and assuming it'd be ok to just make the `title` lowercase. I'm not sure but that seemed a bit English-centric. Maybe there's some gendered-ness or something else that'll mean the title can't just be inserted in the middle of a sentence.

**Review by p-jackson — COMMENTED** [review]
I was able to see the "Manage site" button in the Jetpack Cloud switcher too, so I've fixed that by exposing the `showManageSitesButton` prop further up the component tree.

I made some other changes like removing unnecessary aria labels (the button is already accessible enough to screen readers).

I think this is good to go now, although I did leave a comment about `SidebarScrollSynchronizer` below because I don't know why that change was necessary.

File: client/layout/index.jsx (hunk 2 above)
Comment (danielbachhuber) [reply client/layout/index.jsx]: > do you about why this `SidebarScrollSynchronizer` component needed to be deactivated when in the `sites` layout focus? I removed this line and everything worked as I expected.

I don't, no. Everything works as expected when I remove it too.

I've reverted it with a1085d287b

**Review by danielbachhuber — APPROVED** [review]
Looks good 👍 

I tested in Chrome, Firefox, and Safari on Mac. I also tested with a variety of dashboard color schemes.

The 'Add new site' button ended up stuck to the bottom in non-agency JP Cloud views, so I broadened the CSS override with 9ce12232cb5f80bf02da095e5a953ea1111f2007

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7513767

Thank you @zaguiini for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7513767).

**Pull request merged**